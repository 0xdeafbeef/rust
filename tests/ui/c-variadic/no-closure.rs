#![feature(c_variadic)]
#![crate_type = "lib"]

// Check that `...` in closures is rejected.

const F: extern "C" fn(...) = |_: ...| {};
//~^ ERROR C-variadic type `...` may not be nested inside another type

fn foo() {
    let f = |...| {};
    //~^ ERROR: unexpected `...`

    let f = |_: ...| {};
    //~^ ERROR C-variadic type `...` may not be nested inside another type
    f(1i64)
}
