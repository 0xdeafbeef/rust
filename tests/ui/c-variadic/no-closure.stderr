error[E0743]: C-variadic type `...` may not be nested inside another type
  --> $DIR/no-closure.rs:6:35
   |
LL | const F: extern "C" fn(...) = |_: ...| {};
   |                                   ^^^

error: unexpected `...`
  --> $DIR/no-closure.rs:10:14
   |
LL |     let f = |...| {};
   |              ^^^ not a valid pattern
   |
   = note: only `extern "C"` and `extern "C-unwind"` functions may have a C variable argument list

error[E0743]: C-variadic type `...` may not be nested inside another type
  --> $DIR/no-closure.rs:13:17
   |
LL |     let f = |_: ...| {};
   |                 ^^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0743`.
